src/main/java/com/ocient/jdbc/XGConnection.java

```diff
@@ -2549,7 +2648,8 @@ public void connect() throws Exception
 		connect(this.info.getIp(), this.info.getPortNum());
 		try
 		{
-			clientHandshake(user, pwd, database, false);
+			// FIXME Blocking Call
+			blockingGet(clientHandshake(user, pwd, database, false));
```

Comment:
Prefer `blockingGetSQLResult` for methods that return `CompletionStage<SQLResult<?>>` since `blockingGet(...)` returns an IOException which has an underlying cause of a SQLException. So the catch block would have to inspect the cause of the IOException to ensure an SQLException wasn't marshaled
```suggestion
			blockingGetSQLResult(clientHandshake(user, pwd, database, false)).get();
```

Reply:
EDIT: I actually nerfed this method on your reconnect() branch so no need to change this on this PR actually. 😃